### PR TITLE
chore: e2e add integ_auth_test_cypress_no_ui test

### PR DESCRIPTION
**Pull request by ashwinkumar6**
#### Description of changes
add `integ_auth_test_cypress_no_ui` test

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/callable-e2e-tests-headless.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        working-directory: aws-amplify
+
+      - name: 'Link packages into samples staging root'
+        run: |
+          yarn link amazon-cognito-identity-js @aws-amplify/core @aws-amplify/cache @aws-amplify/auth
+        working-directory: amplify-js-samples-staging
+
+      - name: Run cypress tests for ${{ inputs.test_name }}
+        run: |
+          ../aws-amplify/.circleci/retry-yarn-script.sh -s \
```
Comment (stocaaro) [inline .github/workflows/callable-e2e-tests-headless.yml]: Two things.
1. The `aws-amplify` part of this path shouldn't work anymore, right? Should be `amplify-js` now.
2. As a separate change, can we copy this script and anything else from .circleci into scripts (or somewhere), retaining the `chmod` perm level, but moved so this directory can be deleted?

File: .github/workflows/callable-e2e-tests-headless.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        with:
+          # Minimal depth 2 so we can checkout the commit before possible merge commit.
+          fetch-depth: 2
+          path: aws-amplify
+
+      - name: Setup node and build the repository
+        uses: ./aws-amplify/.github/actions/node-and-build
+
+      - name: Setup samples staging repository
+        uses: ./aws-amplify/.github/actions/setup-samples-staging
```
Comment (stocaaro) [inline .github/workflows/callable-e2e-tests-headless.yml]: Pack fix? `aws-amplify` -> `amplify-js`

File: .github/workflows/callable-e2e-tests-headless.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@24cb9080177205b6e8c946b17badbe402adc938f # v3.4.0 https://github.com/actions/checkout/commit/24cb9080177205b6e8c946b17badbe402adc938f
+        with:
+          # Minimal depth 2 so we can checkout the commit before possible merge commit.
+          fetch-depth: 2
+          path: aws-amplify
+
+      - name: Setup node and build the repository
+        uses: ./aws-amplify/.github/actions/node-and-build
```
Comment (stocaaro) [inline .github/workflows/callable-e2e-tests-headless.yml]: Pack fix? `aws-amplify` -> `amplify-js`

File: .github/workflows/callable-e2e-tests-headless.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+      - name: Setup samples staging repository
+        uses: ./aws-amplify/.github/actions/setup-samples-staging
+        with:
+          GH_PAT: ${{ secrets.GH_PAT }}
+
+      - name: 'Change directory into build and run link'
+        run: |
+          yarn link-all
+        working-directory: aws-amplify
```
Comment (stocaaro) [inline .github/workflows/callable-e2e-tests-headless.yml]: Pack fix? `aws-amplify` -> `amplify-js`

File: .github/workflows/callable-e2e-tests-headless.yml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    name: 'E2E-Headless: ${{ inputs.test_name }}'
+    runs-on: ubuntu-latest
+    timeout-minutes: ${{ inputs.timeout_minutes }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@24cb9080177205b6e8c946b17badbe402adc938f # v3.4.0 https://github.com/actions/checkout/commit/24cb9080177205b6e8c946b17badbe402adc938f
+        with:
+          # Minimal depth 2 so we can checkout the commit before possible merge commit.
+          fetch-depth: 2
+          path: aws-amplify
```
Comment (stocaaro) [inline .github/workflows/callable-e2e-tests-headless.yml]: Pack fix? `aws-amplify` -> `amplify-js`

File: .github/workflows/callable-e2e-tests-headless.yml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+      - name: Setup node and build the repository
+        uses: ./aws-amplify/.github/actions/node-and-build
+
+      - name: Setup samples staging repository
+        uses: ./aws-amplify/.github/actions/setup-samples-staging
+        with:
+          GH_PAT: ${{ secrets.GH_PAT }}
```
Comment (stocaaro) [inline .github/workflows/callable-e2e-tests-headless.yml]: Not sure which change will merge first, but I had been meaning to change this name to something different and went ahead to change it to `GH_ACTIONS_STAGING_READ`

File: .github/workflows/callable-e2e-tests-headless.yml (hunk 2 above)
Comment (ashwinkumar6) [reply .github/workflows/callable-e2e-tests-headless.yml]: updated

File: .github/workflows/callable-e2e-tests-headless.yml (hunk 3 above)
Comment (ashwinkumar6) [reply .github/workflows/callable-e2e-tests-headless.yml]: thx, updated

File: .github/workflows/callable-e2e-tests-headless.yml (hunk 4 above)
Comment (ashwinkumar6) [reply .github/workflows/callable-e2e-tests-headless.yml]: thx, updated

File: .github/workflows/callable-e2e-tests-headless.yml (hunk 5 above)
Comment (ashwinkumar6) [reply .github/workflows/callable-e2e-tests-headless.yml]: thx, updated

File: .github/workflows/callable-e2e-tests-headless.yml (hunk 6 above)
Comment (ashwinkumar6) [reply .github/workflows/callable-e2e-tests-headless.yml]: makes sense, updated

File: .github/workflows/callable-e2e-tests-headless.yml (hunk 1 above)
Comment (ashwinkumar6) [reply .github/workflows/callable-e2e-tests-headless.yml]: sure it'll be good to move `retry-yarn-script.sh` to a different dir. 
let's discuss this

**Review by stocaaro — APPROVED** [review]

**Pull request merged**